app/controllers/web_requests_controller.rb

```diff
@@ -24,7 +26,13 @@ def handle_request
     if user
       agent = user.agents.find_by_id(params[:agent_id])
       if agent
-        content, status, content_type = agent.trigger_web_request(request)
+        content, status, content_type, headers = agent.trigger_web_request(request)
+
+        if headers.presence
+          headers.each do |k,v|
+            response.headers[k] = v
+          end
+        end
```

Comment:
Just asking is it secure to set any header options? do we need a blacklist options in this case ?